### ISSUE TITLE
Decide back-end server design. Java/Javascript

### ISSUE THREAD

**Issue by cheriejw** [PoroRift/pororift-client#11]
REST service is decided but not sure which language/tools.

**Comment by cheriejw** [comment]
Makes more sense to use Node than Java Spring Boot right now, though I want to learn Spring Boot...

**Comment by RichCherng** [comment]
Interesting Read:
https://www.infoworld.com/article/2883328/node-js/java-vs-nodejs-an-epic-battle-for-developer-mindshare.html
https://www.quora.com/What-are-the-advantages-of-Spring-Boot-over-Node-js-for-a-RESTful-web-service
https://stackshare.io/stackups/nodejs-vs-spring
https://dzone.com/articles/nodejs-for-enterprise-applications-are-you-kidding
https://thewebspark.com/2017/12/10/java-spring-or-javascript-node-js/

**Comment by RichCherng** [comment]
TLDR (So far...) Still need more research
Java is more for complex, large scale application
Node is for fast development non SQL project ( GOOD for SPA)

**Comment by RichCherng** [comment]
Outstanding different:
Java:
- more library
- more reliable
- better IDE
- Security (We don't really need this)
- bad with JSON DB (mongoDB) but good with SQL

Node:
- Fast development
- Better unit test (opinionated)
- Simple and easy to maintain
- Good for REST API development
- Bad with SQL but very good with JSON (MongoDB)
- Less Library and solution (Because Node is newer
- Hard to maintain if the software become complex and large scale (Most likely not for this project)

**Comment by cheriejw** [comment]
Yeah, I was leaning towards Javascript because of that. I found those bullet points earlier too.

**Comment by cheriejw** [comment]
Decided to use JavaScript.
